netebpfext/net_ebpf_ext_sock_addr.c

```diff
@@ -0,0 +1,283 @@
+            &_ebpf_sock_addr_hook_provider_moduleid[i],
+            &_net_ebpf_extension_sock_addr_hook_provider_data,
+            EXECUTION_DISPATCH};
+
+        // Set the attach type as the provider module id.
+        _ebpf_sock_addr_hook_provider_moduleid[i].Length = sizeof(NPI_MODULEID);
+        _ebpf_sock_addr_hook_provider_moduleid[i].Type = MIT_GUID;
+        _ebpf_sock_addr_hook_provider_moduleid[i].Guid = *_net_ebpf_extension_sock_addr_attach_types[i];
+
+        // Register the provider context and pass the pointer to the WFP filter parameters
+        // corresponding to this hook type as custom data.
+        status = net_ebpf_extension_hook_provider_register(
+            &hook_provider_parameters,
+            net_ebpf_extension_sock_addr_on_client_attach,
+            _net_ebpf_extension_sock_addr_on_client_detach,
+            &_net_ebpf_extension_sock_addr_wfp_filter_parameters[i],
+            &_ebpf_sock_addr_hook_provider_context[i]);
+    }
+
+    if (status != EBPF_SUCCESS)
+        goto Exit;
+
+Exit:
+    return status;
+}
+
+void
+net_ebpf_ext_sock_addr_unregister_providers()
+{
+    for (int i = 0; i < NET_EBPF_SOCK_ADDR_HOOK_PROVIDER_COUNT; i++)
+        net_ebpf_extension_hook_provider_unregister(_ebpf_sock_addr_hook_provider_context[i]);
+    net_ebpf_extension_program_info_provider_unregister(_ebpf_sock_addr_program_info_provider_context);
+}
+
+//
+// WFP callout callback functions.
+//
+void
+net_ebpf_ext_authorize_connection_classify(
+    _In_ const FWPS_INCOMING_VALUES* incoming_fixed_values,
+    _In_ const FWPS_INCOMING_METADATA_VALUES* incoming_metadata_values,
+    _Inout_opt_ void* layer_data,
+    _In_opt_ const void* classify_context,
+    _In_ const FWPS_FILTER* filter,
+    uint64_t flow_context,
+    _Inout_ FWPS_CLASSIFY_OUT* classify_output)
+{
+    SOCKADDR_IN addr = {AF_INET};
+    uint32_t result;
+    net_ebpf_extension_hook_client_t* attached_client = NULL;
+    bpf_sock_addr_t sock_addr_ctx = {0};
+
+    UNREFERENCED_PARAMETER(incoming_metadata_values);
+    UNREFERENCED_PARAMETER(layer_data);
+    UNREFERENCED_PARAMETER(classify_context);
+    UNREFERENCED_PARAMETER(flow_context);
+
+    attached_client = (net_ebpf_extension_hook_client_t*)filter->context;
+    ASSERT(attached_client != NULL);
+    if (attached_client == NULL)
+        goto Exit;
+
+    if (!net_ebpf_extension_hook_client_enter_rundown(attached_client, EXECUTION_PASSIVE)) {
+        classify_output->actionType = FWP_ACTION_PERMIT;
+        goto Exit;
+    }
+
+    addr.sin_port = incoming_fixed_values->incomingValue[FWPS_FIELD_ALE_AUTH_CONNECT_V4_IP_LOCAL_PORT].value.uint16;
+    addr.sin_addr.S_un.S_addr =
+        incoming_fixed_values->incomingValue[FWPS_FIELD_ALE_AUTH_CONNECT_V4_IP_LOCAL_ADDRESS].value.uint32;
+
+    net_ebpf_extension_hook_invoke_program(attached_client, &sock_addr_ctx, &result);
+
+    classify_output->actionType = FWP_ACTION_PERMIT;
```

Comment:
shouldn't `result` be used here?

Reply:
This is addressed in #891. In this PR the function is basically a placeholder to make compiler happy.